From 5ad97cbc513b3eee3686dd9e85db4a641643e125 Mon Sep 17 00:00:00 2001
From: "Agrawal, Nitesh-kumar" <Nitesh-kumar.Agrawal@amd.com>
Date: Fri, 9 Sep 2016 15:18:09 +0000
Subject: [PATCH 173/198] pinctrl/amd: switch to using a bool for level

commit e084448b5d26bfebe8a7b9c43bb57e685567563d upstream.

The earlier patch can be simplified by using a bool
to indicate level trigger.

Reviewed-by: Pankaj Sen <Pankaj.Sen@amd.com>
Signed-off-by: Nitesh Kumar Agrawal <Nitesh-kumar.Agrawal@amd.com>
[Fixup to earlier manually applied patch]
Signed-off-by: Linus Walleij <linus.walleij@linaro.org>

Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 drivers/pinctrl/pinctrl-amd.c |    9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/pinctrl/pinctrl-amd.c b/drivers/pinctrl/pinctrl-amd.c
index 4a74828..aea310a 100644
--- a/drivers/pinctrl/pinctrl-amd.c
+++ b/drivers/pinctrl/pinctrl-amd.c
@@ -383,7 +383,7 @@ static int amd_gpio_irq_set_type(struct irq_data *d, unsigned int type)
 	int ret = 0;
 	u32 pin_reg;
 	unsigned long flags;
-	u32 level_trig;
+	bool level_trig;
 	u32 active_level;
 	struct gpio_chip *gc = irq_data_get_irq_chip_data(d);
 	struct amd_gpio *gpio_dev = gpiochip_get_data(gc);
@@ -396,13 +396,12 @@ static int amd_gpio_irq_set_type(struct irq_data *d, unsigned int type)
 	 * default settings, ignore incoming settings from client and use
 	 * BIOS settings to configure GPIO register.
 	 */
-	level_trig = pin_reg & (LEVEL_TRIGGER << LEVEL_TRIG_OFF);
+	level_trig = !(pin_reg & (LEVEL_TRIGGER << LEVEL_TRIG_OFF));
 	active_level = pin_reg & (ACTIVE_LEVEL_MASK << ACTIVE_LEVEL_OFF);
 
-	if((!level_trig) &&
-	   ((active_level >> ACTIVE_LEVEL_OFF) == ACTIVE_HIGH)) {
+	if(level_trig &&
+	   ((active_level >> ACTIVE_LEVEL_OFF) == ACTIVE_HIGH))
 		type = IRQ_TYPE_EDGE_FALLING;
-	}
 
 	switch (type & IRQ_TYPE_SENSE_MASK) {
 	case IRQ_TYPE_EDGE_RISING:
-- 
1.7.9.5

